datumaro/plugins/open_images_format.py

```diff
@@ -376,3 +476,42 @@ def _save_subsets(self):
                                 'LabelName': label_categories[annotation.label].name,
                                 'Confidence': str(annotation.attributes.get('score', 1)),
                             })
+                        elif annotation.type is AnnotationType.bbox:
+                            if bbox_description_writer is None:
+                                bbox_description_writer = annotation_writers.enter_context(
+                                    self._open_csv_annotation(
+                                        subset_name + OpenImagesPath.BBOX_DESCRIPTION_FILE_SUFFIX,
+                                        OpenImagesPath.BBOX_DESCRIPTION_FIELDS))
+                                bbox_description_writer.writeheader()
+
+                            if item.has_image:
+                                image_meta[item.id] = (height, width) = item.image.size
+                            else:
+                                log.warning(
+                                    "Can't encode box for item '%s' due to missing image file",
+                                    item.id)
+                                continue
+
+                            bbox_description_writer.writerow({
+                                'ImageID': item.id,
+                                'LabelName': label_categories[annotation.label].name,
+                                'Confidence': str(annotation.attributes.get('score', 1)),
+                                'XMin': annotation.x / width,
+                                'YMin': annotation.y / height,
+                                'XMax': (annotation.x + annotation.w) / width,
+                                'YMax': (annotation.y + annotation.h) / height,
+                                **{
+                                    bool_attr.oid_name:
+                                        int(annotation.attributes.get(bool_attr.datumaro_name, -1))
+                                    for bool_attr in OpenImagesPath.BBOX_BOOLEAN_ATTRIBUTES
+                                },
+                            })
+
+        if image_meta:
+            image_meta_file_path = osp.join(
+                self._save_dir, OpenImagesPath.ANNOTATIONS_DIR,
+                DEFAULT_IMAGE_META_FILE_NAME)
+
+            with open(image_meta_file_path, 'w') as image_meta_file:
+                for image_name, (height, width) in image_meta.items():
+                    print(shlex.quote(image_name), height, width, file=image_meta_file)
```

Comment:
I have a feeling we need to provide a function to write arbitrary lines in this format in addition to the reading function. This will simplify the client code and abstracts it from direct value encoding.

Reply:
Done.